File: UI/travle_core/lib/src/models/tour_schedule_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'tour_schedule_insert_request.g.dart';

/// A new schedule slot for a tour (mirrors the backend `TourScheduleInsertRequest`).
/// The tour comes from the route; the organizer picks [startsAt] only and the end
/// is derived server-side from the tour's duration. [capacity] is optional and
/// defaults to the tour's capacity when omitted.
@JsonSerializable(includeIfNull: false)
class TourScheduleInsertRequest {
  TourScheduleInsertRequest({
    required this.startsAt,
    this.capacity,
  });

  final DateTime startsAt;

  /// Per-slot capacity; when null the tour's default capacity is used.
  final int? capacity;

  factory TourScheduleInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$TourScheduleInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$TourScheduleInsertRequestToJson(this);
}

File: Backend/Travle.Model/Requests/TourScheduleInsertRequest.cs
namespace Travle.Model.Requests
{
    /// <summary>
    /// A new schedule slot for a tour (the tour is taken from the route, never the body). The organizer
    /// picks the start only — the end is derived server-side as <c>StartsAt + tour.DurationMinutes</c>,
    /// so a slot can never contradict the tour's stated duration. <see cref="Capacity"/> is optional and
    /// defaults to the tour's capacity when omitted.
    /// </summary>
    public class TourScheduleInsertRequest
    {
        public DateTime StartsAt { get; set; }

        /// <summary>Per-slot capacity; when null the tour's default capacity is used.</summary>
        public int? Capacity { get; set; }
    }
}
